src/actions/post_build.rs

```diff
@@ -76,16 +71,17 @@ impl PostBuildHandler {
                         t.unpark();
                     }
 
-                    self.notifier.notify_end();
+                    self.notifier.notify_end_diagnostics();
+                    self.active_build_count.fetch_sub(1, Ordering::SeqCst);
                 });
             }
             BuildResult::Squashed => {
                 trace!("build - Squashed");
-                self.notifier.notify_end();
+                self.active_build_count.fetch_sub(1, Ordering::SeqCst);
```

Comment:
I feel like we shouldn't have to manually decrease the count here (possibly by passing the guard value only in `BuildResult::Success`?), but that's outside of the scope of the PR now.